crates/transpiler/src/passes/remove_identity_equiv.rs

```diff
@@ -26,13 +28,182 @@ use qiskit_circuit::packed_instruction::PackedInstruction;
 
 const MINIMUM_TOL: f64 = 1e-12;
 
+/// Fidelity-based computation to check whether an operation `G` is equivalent
+/// to identity up to a global phase.
+///
+/// # Arguments
+///
+/// * `tr_over_dim`: `|Tr(G)| / dim(G)`.
+/// * `dim`: `dim(G)`.
+/// * `error`: tolerance.
+///
+/// # Returns
+///
+/// * `(true, update to the global phase)`` if the operation can be removed.
+/// * `(false, 0.)` if the operation cannot be removed.
+pub fn can_remove(tr_over_dim: Complex64, dim: f64, error: f64) -> (bool, f64) {
+    let f_pro = tr_over_dim.abs().powi(2);
+    let gate_fidelity = (dim * f_pro + 1.) / (dim + 1.);
+    if (1. - gate_fidelity).abs() < error {
+        (true, tr_over_dim.arg())
+    } else {
+        (false, 0.)
+    }
+}
+
+/// Fidelity-based computation to check whether an operation `inst` is equivalent
+/// to identity up to a global phase.
+///
+/// # Arguments
+///
+/// * `inst`: the packed instruction
+/// * `matrix_max_num_qubits`: maximum number of qubits allowed for matrix-based checks.
+/// * `matrix_from_definition`: if `true`, can call the Python-space `Operator` class to
+///   construct the matrix.
+/// * `error_cutoff_fn`: function to compute the allowed error tolerance.
+///
+/// # Returns
+///
+/// * `(true, update to the global phase)`` if the operation can be removed.
+/// * `(false, 0.)` if the operation cannot be removed.
+pub fn is_identity_equiv<F>(
+    inst: &PackedInstruction,
+    matrix_max_num_qubits: Option<u32>,
+    matrix_from_definition: bool,
```

Comment:
Is this ever set to `false`? If it is always `true` I don't think we need to introduce it here (and I think further below it also comes up)